Source/Controllers/DataControllers/SPTableData.m

```diff
@@ -1403,6 +1403,21 @@ - (NSDictionary *) parseFieldDefinitionStringParts:(NSArray *)definitionParts
 			}
 			definitionPartsIndex++;
 
+        // Whether field is generated always as xxxx (VIRTUAL or STORED)
+        } else if ([detailString isEqualToString:@"GENERATED"] && (definitionPartsIndex + 4 < partsArrayLength)
+                    && [[[definitionParts safeObjectAtIndex:definitionPartsIndex+1] uppercaseString] isEqualToString:@"ALWAYS"]
+                    && [[[definitionParts safeObjectAtIndex:definitionPartsIndex+2] uppercaseString] isEqualToString:@"AS"]) {
+            if ([[aValue = [definitionParts safeObjectAtIndex:definitionPartsIndex+4] uppercaseString] isEqualToString:@"VIRTUAL"]) {
+                [fieldDetails setValue:aValue forKey:@"generatedalways"];
+                [[fieldDetails objectForKey:@"unparsed"] appendString:@" "];
+                [[fieldDetails objectForKey:@"unparsed"] appendString:[definitionParts safeObjectAtIndex:definitionPartsIndex+3]];
+            } else if ([[aValue = [definitionParts safeObjectAtIndex:definitionPartsIndex+4] uppercaseString] isEqualToString:@"STORED"]) {
+                [fieldDetails setValue:aValue forKey:@"generatedalways"];
```

Comment:
Correct me if I'm wrong, but if you are just storing the result of this boolean isEqual comparison into fieldDetails, couldn't you eliminate `aValue = ` and just use `YES` in the setValue call?

Reply:
`aValue` does not contain the boolean value of isEqual comparison but the value from the `definitionParts` which is VIRTUAL or STORED.